### ISSUE TITLE
Floating Window Title Restrictions not working in Intellij.

### ISSUE THREAD

**Issue by nliechti** [ianyh/Amethyst#1337]
**Describe the bug**
Popup windows in Intellij are not working with the "Window Title Restrictions". There are not floating, even if excluded in the settings.

**Applications:**
Intellij

**To Reproduce**
For example: Open the Push dialog of Intellij

**Expected behavior**
If "Push Commits" is excluded, the push dialog should be floating

**Screenshots**

**Versions:**
 - macOS: 13.0 Ventura
 - Amethyst: 0.16.0
 
**Debug Info**
```
Version: 0.16.0 (94)

OS version: Version 13.0 (Build 22A380)

Screens:
	(0.0, 0.0, 3840.0, 1600.0) [(0.0, 0.0, 3840.0, 1600.0)]
	(3840.0, 179.0, 1728.0, 1117.0) [(3840.0, 304.0, 1728.0, 1117.0)]

Configuration:
follow-space-thrown-windows: 0
restore-layouts-on-launch: 1
window-margin-size: 2
debug-layout-info: 1
screen-padding-left: 0
enables-layout-hud-on-space-change: 1
window-resize-step: 5
mod1: (
    option,
    shift
)
focus-follows-mouse: 0
smart-window-margins: 1
screen-padding-top: 0
floating-is-blacklist: 1
new-windows-to-main: 0
layouts: (
    "middle-wide",
    tall,
    fullscreen,
    column,
    "3column-left",
    floating
)
window-minimum-height: 0
floating: (
        {
        id = "com.bitwarden.desktop";
        "window-titles" =         (
        );
    },
        {
        id = "org.keepassxc.keepassxc";
        "window-titles" =         (
        );
    },
        {
        id = "com.postbox-inc.postbox";
        "window-titles" =         (
            "Compose: .*"
        );
    },

        {
        id = "com.cisco.Jabber";
        "window-titles" =         (
        );
    },
        {
        id = "com.teamviewer.TeamViewer";
        "window-titles" =         (
        );
    },
        {
        id = "com.readdle.PDFExpert-Mac";
        "window-titles" =         (
        );
    },
        {
        id = "com.markmcguill.strongbox";
        "window-titles" =         (
        );
    },
        {
        id = "com.wireguard.macos";
        "window-titles" =         (
        );
    },
        {
        id = "com.apple.Preview";
        "window-titles" =         (
        );
    },
        {
        id = "com.apple.systempreferences";
        "window-titles" =         (
        );
    },
        {
        id = "com.1password.1password";
        "window-titles" =         (
        );
    },
        {
        id = "org.jitsi.jitsi-meet";
        "window-titles" =         (
        );
    },
        {
        id = "at.obdev.littlesnitch";
        "window-titles" =         (
        );
    },
        {
        id = "com.jetbrains.toolbox.linkapp.pcom.jetbrains.toolbox__ntelli_ltimate_2022_3_eta";
        "window-titles" =         (
            "Push Commits"
        );
    }
)
window-margins: 1
window-minimum-width: 0
screen-padding-right: 0
mouse-swaps-windows: 1
screen-padding-bottom: 0
mouse-follows-focus: 0
float-small-windows: 1
use-canary-build: 0
mouse-resizes-windows: 1
ignore-menu-bar: 0
mod2: (
    option,
    shift,
    control
)
enables-layout-hud: 1


```
Note: `--include-apps` will list your manageable applications, but is optional if you don't want to list that.

**Comment by usmajeed** [comment]
Same with Pycharm on Ventura.

**Comment by devployment** [comment]
I'd suggest to use the bundle-id instead of the toolbox app name. For example in my case, for Rider it would be something like this, just getting the bundle id of the running instance. 

`com.jetbrains.toolbox.linkapp.pcom.jetbrains.toolbox__ider_2022_3`

This is what ` ~/Applications/JetBrains Toolbox/Rider.app` is exposing. For what I know this is just a "wrapper", as the underlying app is managed by the toolbox.

These underlying apps are located at `~/Library/Application Support/JetBrains/Toolbox` (by default?). This path can be configured in the toolbox app itself. 

In this folder you will find several folders, containing sub-folders with the several versions installed. If you browse the contents of the `*.app` you'll find a `Info.plist` file. That file contains the "real" bundle id you might want to use.

![image](https://user-images.githubusercontent.com/7137845/206983697-3d5f0f82-be0e-43fe-a85e-fa82f75733b9.png)

![image](https://user-images.githubusercontent.com/7137845/206983738-6bc367be-24dd-4879-97c2-8e604cc4fdc4.png)

tl;dr; do not use the bundle ID that's coming from the app started by the toolbox. Use the "real" bundle id. 

At least this is what worked for me.

**Comment by nliechti** [comment]
@devployment Should have come to this solution myself 🤦‍♂️. Thanks for the pointer, works now.
